test/models/user_test.rb

```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class UserTest < ActiveSupport::TestCase
```

Comment:
I think we could add some tests here for the `User.from_omniauth(access_token)` method. Maybe one that checks that a user gets created if one with the same email doesn't exist already. And one that checks the opposite.

There are some helper methods which could be helpful here, e.g. `assert_difference, User.count, 1` - this checks that a new user record gets created. The opposite can be done as well, e.g:

(Not tested btw, not sure if this runs fully)
```
test 'a user with matching email address exists` do
  user = users(:schmidt) 
  assert_no_difference 'User.count' do
    matching_user = User.from_omniauth({'email' => user.email})

    assert_equal user, matching_user
  end 
end
```

Reply:
Thank you for your help! Done.